### PR TITLE
Update docs for 1.5

### PR DESCRIPTION
**Pull request by raoulcollenteur**
# Short Description

PR to make a small changes to the documentation for 1.5, mostly the header menu as requested by @mbakker7. 

# Checklist before PR can be merged:
- [x] closes issue #737
- [x] is documented
- [x] Format code with [Black formatting](https://black.readthedocs.io)
- [x] type hints for functions and methods
- [x] tests added / passed
- [x] Example Notebook (for new features)
- [x] Remove output for all notebooks with changes

### CODE REVIEW

**Comment by mbakker7** [comment]
Is this still going to be a change for 1.5 (title of PR)?

File: doc/index.rst [hunk 1]
```diff
@@ -99,8 +99,8 @@ you to cite the Pastas article published in Groundwater journal as follows:
 
     User Guide <userguide/index>
     Examples <examples/index>
-    API Docs <api/index>
+    Reference <api/index>
```
Comment (martinvonk) [inline doc/index.rst:102]: Maybe we can call this API Reference just like like Pandas/Numpy/Scipy?

**Review by martinvonk — APPROVED** [review]

**Review by dbrakenhoff — APPROVED** [review]

**Comment by mbakker7** [comment]
What I found confusing is that the menu at the top list Benchmarks but the 6 boxes don't list Benchmarks. I (and others like @rubencalje ) had a difficult time finding the benchmark notebooks.

File: doc/index.rst (hunk 1 above)
Comment (raoulcollenteur) [reply doc/index.rst:102]: Good idea will do, same for the other item (contribute > development)

**Comment by raoulcollenteur** [comment]
> Is this still going to be a change for 1.5 (title of PR)?

Yes that is the idea, some small improvements.

> What I found confusing is that the menu at the top list Benchmarks but the 6 boxes don't list Benchmarks. I (and others like @rubencalje ) had a difficult time finding the benchmark notebooks. This PR doesn't seem to fix that, so it doesn't really close #737. Other than that it looks good.

I will update the issue name of @rubencalje on the benchmark notebooks being hard to find (although in the menu). To be discussed wrt NWO project.

**Comment by rubencalje** [comment]
Do you mean my comments in issue #705? I did not say the benchmark notebooks were hard to find. I mentioned there is a section about the WellModel that used to be somewhere in the documentation, that would be nice to be placed under 'benchmarks', and that I would add this again.

**Comment by mbakker7** [comment]
Ok. Sorry I misunderstood.

**Pull request merged**